cpp/include/cudf/detail/reverse.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/types.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+#include <memory>
+
+namespace cudf {
+namespace detail {
+
+/**
+ * @copydoc cudf::reverse(table_view const&,
+ * rmm::mr::device_memory_resource*)
+ *
+ * @param stream CUDA stream used for device memory operations and kernel launches.
+ */
+std::unique_ptr<table> reverse(
+  table_view const& source_table,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
```

Comment:
Detail APIs should take streams (if they do anything that can run on a stream).
```suggestion
std::unique_ptr<table> reverse(
  table_view const& source_table,
  rmm::cuda_stream_view stream = rmm::cuda_stream_view{},
  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
```

Reply:
Should this be `rmm::cuda_stream_view stream = rmm::cuda_stream_default`? This seems to be the case in `repeat.hpp`.

Reply:
> Should this be `rmm::cuda_stream_view stream = rmm::cuda_stream_default`? This seems to be the case in `repeat.hpp`.

A default constructed `cuda_stream_view` is the same as `cuda_stream_default`, so it doesn't really matter which you use.